html/pfappserver/lib/pfappserver/Authentication/Store/PacketFence/User.pm

```diff
@@ -46,23 +46,33 @@ sub check_password {
 
   my $internal_sources = pf::authentication::getInternalAuthenticationSources();
   my ($stripped_username,$realm) = strip_username($self->_user);
-  my $realm_source = get_realm_authentication_source($stripped_username, $realm);
-  if($realm_source && any {$_->id eq $realm_source->id} @{$internal_sources}){
-    get_logger->info("Found realm source ".$realm_source->id." for user $stripped_username in realm $realm. Using it as the only source.");
-    $internal_sources = [$realm_source];
+  my $realm_source = get_realm_authentication_source($stripped_username, $realm, $internal_sources);
+
+  my $user = $self->{_user};
+  if (ref($realm_source) eq 'ARRAY') {
+    foreach my $source (@{$realm_source}) {
+      get_logger->info("Found realm source ".$source->id." for user $stripped_username in realm $realm. Using it as the only source.");
```

Comment:
That log message isn't really right, its not being used as the «only» source since you're looping on the realm_source array which may contain more than 1